apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -544,6 +618,8 @@ private Publisher<? extends Result> insert(List<Span> spans, Connection connecti
             int i = 0;
             for (Span span : spans) {
 
+                BigDecimal estimatedCost = calculateCost(span);
```

Comment:
This is mixing responsibilities. I think all these cost calculations have to be done in the service level prior to passing the spans here. The DAO should just take care of storing and retrieving, without business logic.

Reply:
I missed previous comment about this.